miner/worker.go

```diff
@@ -859,23 +883,33 @@ func (w *worker) commitNewWork(interrupt *int32, noempty bool, timestamp int64)
 		misc.ApplyDAOHardFork(env.state)
 	}
 	// Accumulate the uncles for the current block
-	for hash, uncle := range w.possibleUncles {
+	for hash, uncle := range w.localUncles {
 		if uncle.NumberU64()+staleThreshold <= header.Number.Uint64() {
-			delete(w.possibleUncles, hash)
+			delete(w.localUncles, hash)
 		}
 	}
-	uncles := make([]*types.Header, 0, 2)
-	for hash, uncle := range w.possibleUncles {
-		if len(uncles) == 2 {
-			break
+	for hash, uncle := range w.remoteUncles {
+		if uncle.NumberU64()+staleThreshold <= header.Number.Uint64() {
+			delete(w.remoteUncles, hash)
 		}
-		if err := w.commitUncle(env, uncle.Header()); err != nil {
-			log.Trace("Possible uncle rejected", "hash", hash, "reason", err)
-		} else {
-			log.Debug("Committing new uncle to block", "hash", hash)
-			uncles = append(uncles, uncle.Header())
+	}
+	uncles := make([]*types.Header, 0, 2)
+	commitUncle := func(blocks map[common.Hash]*types.Block) {
```

Comment:
`commitUncle` -> `commitUncles`

Might as well move the uncle cleaning loop in here too, removes some code duplication.